src/SDK.ts

```diff
@@ -281,9 +281,9 @@ export function init(options?: IExtensionInitOptions): Promise<void> {
         const initOptions = { ...options, sdkVersion };
 
         parentChannel.invokeRemoteMethod<IExtensionHandshakeResult>("initialHandshake", hostControlId, [initOptions]).then((handshakeData) => {
-            hostPageContext = handshakeData?.pageContext;
-            webContext = hostPageContext?.webContext;
-            teamContext = webContext?.team;
+            hostPageContext = handshakeData.pageContext;
+            webContext = hostPageContext.webContext;

```

Comment:
since `hostPageContext` could be undefined, we should guard against that
```
hostPageContext = handshakeData.pageContext;
webContext = handshakeData ? handshakeData.webContext : undefined;
teamContext = webContext ? webaContext.team : undefined;
```

Reply:
(also: is my understanding correct that the coalesce operator `?.` is the thing we're trying to remove here?  to solve a typescript version issue?)

Reply:
Yes. We are trying to solve a typescript version issue here. Upgrading it to version `3.*.*` above, will result in more build errors.